src/frontend/next/src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,18 @@
+import PostSearchInput from './PostSearchInput';
+import AuthorSearchInput from './AuthorSearchInput';
+
+type searchInputProps = {
+  text: string;
+  onTextChange: Function;
```

Comment:
same question here as in #1581 , should we be more specific about the type here? Maybe `EventHandler` instead of `Function`

Reply:
It's an onChange event so ChangeEvent is the type here, same as in #1581

Reply:
Let's do it in a follow-up PR, because we are drilling a lot of stuff here. After getting everything working we will need to create an interface to `SearchPage`'s components and a `context.` These functions are connected to how we build the URL query of a search. So I believe that we need to be care full with that. It's working now. Let's keep this for now. 

I'm opening an issue for the next step (interface and context) now.

Reply:
https://github.com/Seneca-CDOT/telescope/issues/1584

Reply:
Sounds good